src/common/browser-adapters/browser-adapter-event-manager.ts

```diff
@@ -22,24 +24,21 @@ const TWO_MINUTES = 120000;
 const FOUR_MINUTES = 2 * TWO_MINUTES;
 
 export class BrowserAdapterEventManager {
-    protected deferredEvents: EventDetails[] = [];
-    protected eventsToApplicationListenersMapping: DictionaryStringTo<ApplicationListener[]> = {};
+    protected deferredEvents: DeferredEventDetails[] = [];
+    protected eventsToApplicationListenersMapping: DictionaryStringTo<ApplicationListener> = {};
     protected adapterListener: AdapterListener =
         (eventType: string) =>
         (...eventArgs: any[]) => {
             this.processEvent(eventType, eventArgs);
         };
-    constructor(private promiseFactory: PromiseFactory, private timeoutFactory: TimeoutFactory) {}
+
+    constructor(private promiseFactory: PromiseFactory) {}
 
     public registerEventToApplicationListener = (
         eventType: string,
         callback: ApplicationListener,
     ): void => {
-        if (this.eventsToApplicationListenersMapping[eventType]) {
-            this.eventsToApplicationListenersMapping[eventType].push(callback);
-        } else {
-            this.eventsToApplicationListenersMapping[eventType] = [callback];
-        }
+        this.eventsToApplicationListenersMapping[eventType] = callback;
```

Comment:
Maybe have this be defensive about multiple registration (ie, detect and throw, rather than silently overriding)?